labs-ios-starter/Map/MapViewController.swift

```diff
@@ -8,42 +8,117 @@
+    let regionInMeters: Double = 10000
+    
+    var resultSearchController: UISearchController? = nil
+    var selectedPin: MKPlacemark? = nil
+    
     override func viewDidLoad() {
         super.viewDidLoad()
         
-        //Currently set to San Francisco - change to user location
-        let initialLocation = CLLocation(latitude: 37.773972, longitude: -122.431297)
-        mapView.centerToLocation(initialLocation)
+        checkLocationServices()
+        
+        // For the searchController (will set in the navigation bar)
+        let locationSearchTable = storyboard!.instantiateViewController(identifier: "LocationSearchTable") as! LocationSearchTableViewController
+        resultSearchController = UISearchController(searchResultsController: locationSearchTable)
+        resultSearchController?.searchResultsUpdater = locationSearchTable
+        
+        let searchBar = resultSearchController!.searchBar
+        searchBar.sizeToFit()
+        searchBar.placeholder = "Search for location"
+        navigationItem.titleView = resultSearchController?.searchBar
+        
+        resultSearchController?.hidesNavigationBarDuringPresentation = false
+        definesPresentationContext = true
+        
+        //Passes the mapView to the SearchTable
+        locationSearchTable.mapView = mapView
+        
+        locationSearchTable.handleMapSearchDelegate = self
+
     }
     
+    func setupLocationManager() {
+        locationManager.delegate = self
+        locationManager.desiredAccuracy = kCLLocationAccuracyBest
+    }
     
+    func checkLocationServices() {
+        if CLLocationManager.locationServicesEnabled() {
+            setupLocationManager()
+            checkLocationAuthorization()
+        } else {
+            let alert = UIAlertController(title: "Location Permissions Denied", message: "Go to your settings and allow this app to use your location", preferredStyle: .alert)
+            alert.addAction(UIAlertAction(title: "OK", style: .default, handler: nil))
+            self.present(alert, animated: true)
+        }
+    }
     
+    func centerViewOnUserLocation() {
+        if let location = locationManager.location?.coordinate {
+            let region = MKCoordinateRegion.init(center: location, latitudinalMeters: regionInMeters, longitudinalMeters: regionInMeters)
+            mapView.setRegion(region, animated: true )
+        }
+    }
+    
+    func checkLocationAuthorization() {
+        switch CLLocationManager.authorizationStatus() {
+        case .authorizedWhenInUse:
+            mapView.showsUserLocation = true
+            centerViewOnUserLocation()
+        case .denied:
+            let alert = UIAlertController(title: "Location Permissions Denied", message: "Go to your settings and allow this app to use your location", preferredStyle: .alert)
+            alert.addAction(UIAlertAction(title: "OK", style: .default, handler: nil))
+            self.present(alert, animated: true)
```

Comment:
There is actually a way to deep link to the settings app, so that when they hit one of the actions it will take them right into settings. See `UIApplication.openSettingsURLString`